server/handlers.go

```diff
@@ -87,21 +88,38 @@ func (server *Server) generateHandleWS(ctx context.Context, cancel context.Cance
 	}
 }
 
+func (server *Server) closeWSConn(conn *websocket.Conn, code int, reason string) {
+	go func() {
+		for {
+			// Just make websocket.Conn to treat control messages correctly until connection close.
+			_, _, err := conn.ReadMessage()
```

Comment:
This is in accordance to gorilla websocket package documentation. To be sure we handle all control messages, we have to run our own goroutine with .ReadMessage() calls, because we can't guarantee someone calls it at the moment of connection termination.